### PR TITLE
Added option for preventing the checking of parent categories

### PR DESCRIPTION
**Pull request by Defcon0**
Hi @qzminski ,

since we needed this feature, we added it to your module and would be happy if you accepted it. It's an additional option for disabling inputs of parent categories. In our case, it was only possible to select "leaf items".

Hoping for a merge ;-)

Bye Defcon0

### CODE REVIEW

File: widgets/WidgetTreeSelector.php [hunk 1]
```diff
@@ -464,6 +464,13 @@ protected function renderItemTree($id, $intMargin, $blnNoRecursion=false)
         return $return;
     }
 
+    protected function isLeaf($objItem)
```
Comment (qzminski) [inline widgets/WidgetTreeSelector.php]: Can you add a short docblock what this method does?

Comment (qzminski) [reply widgets/WidgetTreeSelector.php]: Or maybe even rename it to something more appropriate, like e.g. `hasChildren`.

File: widgets/WidgetTreeSelector.php [hunk 2]
```diff
@@ -464,6 +464,13 @@ protected function renderItemTree($id, $intMargin, $blnNoRecursion=false)
...
+    {
+        $objChildren = $this->Database->prepare("SELECT id FROM $this->foreignTable WHERE pid=?")->execute($objItem->id);
+
+        return $objChildren->numRows <= 0;
```
Comment (qzminski) [inline widgets/WidgetTreeSelector.php]: Can the `numRows` be ever negative? Please change to: `return (int) $objChildren->numRows === 0;`

**Review by qzminski — CHANGES_REQUESTED** [review]

**Comment by Defcon0** [comment]
@qzminski OK, just fixed according to your request.

**Comment by qzminski** [comment]
Wonderful, thank you!

**Comment by qzminski** [comment]
Just released 2.4.0 👌🏻

**Comment by Defcon0** [comment]
Awesome, thanks ;-)

**Pull request merged**